packages/modules/vehicles/kia/api.py

```diff
@@ -0,0 +1,860 @@
+            f.write(json.dumps(token))
+    except Exception:
+        log.debug("kia.saveToken: token could not be saved", exc_info=True)
+        raise
+
+# ---------- authentication ----------
+
+
+def getCookies(brand: str) -> dict:
+    log.info("Kia/Hyundai: Create Login-session")
+
+    try:
+        url = getString("base_url", brand) + \
+              '/api/v1/user/oauth2/authorize?' + \
+              'response_type=code&state=test&client_id=' + \
+              getString("client_id", brand) + '&redirect_uri=' + \
+              getString("base_url", brand) + '/api/v1/user/oauth2/redirect'
+        cookies = getHTTPCookies(url)
+
+        url = getString("base_url", brand) + '/api/v1/user/session'
+        getHTTP(url=url, cookies=cookies)
+
+        url = getString("base_url", brand) + '/api/v1/user/language'
+        headers = {'Content-type': 'application/json'}
+        data = {"lang": "en"}
+        response = postHTTP(url=url, data=data,
+                            headers=headers, cookies=cookies)
+
+        url = getString("base_url", brand) + '/api/v1/user/session'
+        deleteHTTP(url=url, cookies=cookies)
+    except Exception:
+        log.debug("kia.getCookies: " + response, exc_info=True)
+        raise
+
+    return cookies
+
+
+def getDeviceId(brand: str) -> dict:
+    log.info("Kia/Hyundai: Requesting device ids")
+
+    token = {}
+
+    try:
+        token["gcmClientId"] = str(uuid.uuid4())
+        token["gcmVehicleId"] = str(uuid.uuid4())
+
+        url = getString("base_url", brand) + \
+            '/api/v1/spa/notifications/register'
+        data = {
+            "pushRegId": getString("gcm_sender_id", brand),
+            "pushType": "GCM",
+            "uuid": str(uuid.uuid4())
+            }
+        headers = {
+            'Authorization': '',
+            'Ccsp-Device-Id': '',
+            'Ccsp-Service-Id': getString("client_id", brand),
+            'Ccsp-Application-Id': getString("app_id", brand),
+            'Offset': '2',
+            'Clientid': token["gcmClientId"],
+            'Vehicleid': token["gcmVehicleId"],
+            'Ccuccs2protocolsupport': '0',
+            'Content-type': 'application/json; charset=UTF-8',
+            'Content-Length': str(len(data)),
+            'Host': getString("host", brand),
+            'Connection': 'Keep-Alive',
+            'Accept-Encoding': 'gzip, deflate',
+            'User-Agent': 'okhttp/3.12.12',
+            'Stamp': getStamp(brand)
+            }
+        response = postHTTP(url=url, data=data, headers=headers)
+
+        response_dict = json.loads(response)
+        token["deviceId"] = response_dict['resMsg']['deviceId']
+
+        log.debug("Kia/Hyundai: DeviceId = " + token["deviceId"][:8] + "[...]")
+    except Exception:
+        log.error("kia.getDeviceId: Request device id failed")
+        log.debug("kia.getDeviceId: Request device id failed: " +
+                  response, exc_info=True)
```

Comment:
```suggestion
        log.exception("kia.getDeviceId: Request device id failed")
```
Exceptions bitte immer loggen. Mit log.exception wird auch der Stack automatisch mitgeloggt. Bitte auch an den anderen Stellen anpassen.

Reply:
Ist entsprechend angepasst